### PR TITLE
[Merged by Bors] - Limit to one concurrent on staging

### PR DESCRIPTION
**Pull request by pigmej**
Limit to one concurrent run on staging and in the same time allow multiple on other branches.
Besides, that cancel concurrent IF they're NOT in the staging.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -268,7 +268,9 @@ jobs:
       - filter-changes
       - dockerpush
     timeout-minutes: 90
-    concurrency: ${{ github.workflow }}-${{ github.ref }}
+    concurrency:
+      group: ${{ github.base_ref == 'staging' && format('{0}-staging', github.workflow) || format('{0}-{1}', github.workflow, github.ref) }}
+      cancel-in-progress: ${{ github.base_ref == 'staging' && false || true }}
```
Comment (poszu) [inline .github/workflows/ci.yml:273]: What is going on here: 
```
&& false || true
```

Comment (pigmej) [reply .github/workflows/ci.yml:273]: That's a github workaround for NOT having ternary operator... 
it's basically `condition && if-true || if-false`.

**Review by poszu — APPROVED** [review]

**Comment by pigmej** [comment]
bors try

**Comment by pigmej** [comment]
bors try

**Comment by pigmej** [comment]
bors try

**Comment by pigmej** [comment]
bors merge

**Review by andres-spacemesh — APPROVED** [review]

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**